src/services/anchor-service.ts

```diff
@@ -179,8 +181,18 @@ export class AnchorService {
    */
   // TODO: Remove for CAS V2 as we won't need to move PENDING requests to ready. Switch to using anchorReadyRequests.
   async anchorRequests(abortOptions?: AbortOptions): Promise<boolean> {
+    const timeout = setTimeout(() => {
+      Metrics.record(METRIC_NAMES.ANCHOR_TAKING_TOO_LONG, 1)
+    }, this.alertOnLongAnchorMs)
+
+    abortOptions?.signal?.addEventListener('abort', () => {
+      clearTimeout(timeout)
+    })
+
     if (this.useQueueBatches) {
-      return this.anchorNextQueuedBatch(abortOptions)
+      const results = await this.anchorNextQueuedBatch(abortOptions)
+      clearTimeout(timeout)
+      return results
```

Comment:
Since the result is a boolean for when a batch was anchored, should we rename the variable?
```suggestion
      const batchAnchored = await this.anchorNextQueuedBatch(abortOptions)
      clearTimeout(timeout)
      return batchAnchored
```